x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Eval.java

```diff
@@ -69,6 +71,16 @@ public List<Attribute> output() {
         return lazyOutput;
     }
 
+    @Override
+    protected AttributeSet computeReferences() {
+        return requiredAttributesFromChild(fields);
+    }
+
+    public static AttributeSet requiredAttributesFromChild(List<Alias> fields) {
+        AttributeSet generated = new AttributeSet(asAttributes(fields));
+        return Expressions.references(fields).subtract(generated);
+    }
```

Comment:
Same remark - is this used elsewhere externally?

Reply:
Yup, in `EvalExec`!

Reply:
Any reason why the methods across Eval and Aggregate don't have the same name?  Something like doComputeReferences() or determineReferences(), etc... required is a misnomer in this context.

Reply:
I'll just overload `computeReferences` (and `output` for `Aggregate`), that should be nice and consistent.